### DISCUSSION TITLE
ToolkitProvider default applyClassesTo should always be 'scope'

### DISCUSSION THREAD

**Discussion by heswell** [jpmorganchase/salt-ds#841] [Feedback]
If user doesn't explicitly configure the `applyClassersTo` prop, then the top level ToolkitProvider will default to `root`, nested Providers will default to `scope`. I think the default should always be `scope`, here's why:

In `root` mode, classnames are applied to the document element in a `useLayoutEffect` hook. If any other components in the app also rely on a useLayoutEffect hook, these will execute first, it's bottom-up order. If any of these are using the useLayoutEffect to measure rendered items (a very common use case), they will be doing so, first time around, before the theme classnames have been applied to the document element  so no theme variables will be in scope and measurements are highly likely to be wrong.

If there is a single child under the root Provider, I'd suggest that the default would be to apply the theme classnames there.

**Comment by mark-tate** [comment]
We are happy with the default being `root` as per most other component libraries,  measuring elements via `useLayoutEffect` should be continuous. We are closing out existing discussions, so feel free to create as an `issue` if you want to continue the discussion.
